lib/src/core/selection/selection.ts

```diff
@@ -6,202 +6,198 @@
+    private _deselectedToEmit: T[] = [];
+
+    /** Keeps track of the selected options that haven't been emitted by the change event. */
+    private _selectedToEmit: T[] = [];
+
+    /** Cache for the array value of the selected items. */
+    private _selected: T[] | null;
+
+    /** Selected values. */
+    get selected(): T[] {
+        if (!this._selected) {
+            this._selected = Array.from(this._selection.values());
+        }
+
+        return this._selected;
+    }
+
+    /** Event emitted when the value has changed. */
+    changed: Subject<SelectionChange<T>> = new Subject();
+
+    /**
+     * Event emitted when the value has changed.
+     * @deprecated Use `changed` instead.
+     * @breaking-change 8.0.0 To be changed to `changed`
+     */
+    onChange: Subject<SelectionChange<T>> = this.changed;
+
+    constructor(private _multiple = false, initiallySelectedValues?: T[], private _emitChanges = true) {
+        if (initiallySelectedValues && initiallySelectedValues.length) {
+            if (_multiple) {
+                initiallySelectedValues.forEach(value => this._markSelected(value));
+            } else {
+                this._markSelected(initiallySelectedValues[0]);
+            }
+
+            // Clear the array in order to avoid firing the change event for preselected values.
+            this._selectedToEmit.length = 0;
+        }
+    }
+
+    /**
+     * Selects a value or an array of values.
+     */
+    select(...values: T[]): void {
+        this._verifyValueAssignment(values);
+        values.forEach(value => this._markSelected(value));
+        this._emitChangeEvent();
+    }
+
+    /**
+     * Deselects a value or an array of values.
+     */
+    deselect(...values: T[]): void {
+        this._verifyValueAssignment(values);
+        values.forEach(value => this._unmarkSelected(value));
+        this._emitChangeEvent();
+    }
+
+    /**
+     * Toggles a value between selected and deselected.
+     */
+    toggle(value: T): void {
+        this.isSelected(value) ? this.deselect(value) : this.select(value);
+    }
+
+    /**
+     * Clears all of the selected values.
+     */
+    clear(): void {
         this._unmarkAll();
-      }
+        this._emitChangeEvent();
+    }
+
+    /**
+     * Determines whether a value is selected.
+     */
+    isSelected(value: T): boolean {
+        return this._selection.has(value);
+    }
+
+    /**
+     * Determines whether the model does not have a value.
+     */
+    isEmpty(): boolean {
+        return this._selection.size === 0;
+    }
+
+    /**
+     * Determines whether the model has a value.
+     */
+    hasValue(): boolean {
+        return !this.isEmpty();
+    }
+
+    /**
+     * Sorts the selected values based on a predicate function.
+     */
+    sort(predicate?: (a: T, b: T) => number): void {
+        if (this._multiple && this.selected) {
+            this._selected && this._selected.sort(predicate);
```

Comment:
i beliece we should start using this syntax
```suggestion
           this._selected?.sort(predicate);
```

Reply:
I know but the Typescript version we have in this project doesn't recognize this syntax. I don't want to go through the process of updating Angular at the moment. At least not in this PR.

Reply:
I've seen this syntax in some other file lib/src/modal/modal.ts